### DISCUSSION TITLE
Headers cascading to other sections that don't have headers defined - Using 6.0.3

### DISCUSSION THREAD

**Discussion by ccapps0** [dolanmiu/docx#870] [Q&A] [answered]
Hello, 

I'm running into a weird "issue"...or dev error... I am adding a section that contains a header and then adding another without a header defined. The header defined in the first section is showing up in the second section where no header is defined. The second section should not have a header in it.

Imagine constants table1, paragraph1, table2, paragraph2, and header1. See example below: 

doc.addSection({
  children: [table1, paragraph1]
  headers: header1
});

doc.addSection({ // the header defined in the section above is showing up in this section.
  children: [table2, paragraph2]
});

Is this behavior correct?

**Comment by dolanmiu** [comment, marked as answer]
Seems like this is a bug yes

Strange

A work around would be to make a blank header/footer

**Reply by dolanmiu** [reply]
It's probably intended behavior actually

In `docx`, it does not even add a header in the second section, which I suppose Word tries to be smart and takes the last known header set

**Reply by ccapps0** [reply]
Yeah, I am currently using a workaround that entails defining a blank header, although a paragraph ends up there, setting the run size to 2 makes it super small that it isn't noticeable. 

Another thing to note; this didn't happen in docx 5.5.

**Reply by dolanmiu** [reply]
In `docx` `5.5.0` it intentionally sets a blank header, which was removed in PR:

https://github.com/dolanmiu/docx/pull/829

To address:

https://github.com/dolanmiu/docx/issues/251

Maybe this was the change which broke it?

**Reply by ccapps0** [reply]
Embarrassingly, I was using docx 5.4.1, not 5.5. I skipped 5.5 and went to 6 (now 6.0.3). 

I see the workaround in #251 where they set the margins to null in the overall document. That seems like a **better** solution to my problem versus setting the run size to 2. 

Thanks for getting me informed.

**Reply by ccapps0** [reply]
Actually... I'm not seeing a way to set header margins. Was this removed?

**Reply by dolanmiu** [reply]
Not removed, you can see it being done here:

https://github.com/dolanmiu/docx/blob/master/demo/59-header-footer-margins.ts

It looks a bit unintuiative, but this is how OOXML and Microsoft Word does it

**Reply by ccapps0** [reply]
Thanks!
